compiler/noirc_frontend/src/elaborator/mod.rs

```diff
@@ -1306,8 +1313,10 @@ impl<'context> Elaborator<'context> {
             let unresolved_type = &trait_impl.object_type;
             let self_type_span = unresolved_type.span;
             self.add_generics(&trait_impl.generics);
+            trait_impl.resolved_generics = self.generics.clone();
```

Comment:
I mentioned this in the other PR review, but`recover_generics` could be used here?

Reply:
This case is simpler with just `generics.clear()` I think but it'd be the same either way